assets/controllers/ace_controller.js

```diff
@@ -0,0 +1,144 @@
+import {Controller} from '@hotwired/stimulus';
+import Ace from 'ace-builds/src-noconflict/ace.js';
+import 'ace-builds/src-noconflict/ext-language_tools.js';
+
+/**
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers
+ * @see https://symfony.com/doc/current/frontend/asset_mapper.html#how-does-the-importmap-work
+ * Worker is standard JS build in feature for running jobs on background that use ace editor for loading workers.
+ * We need override this class to support script from importmap generated from symfony/asset-mapper.
+ * So when worker is created with script url, we need to check if this script is in importmap and replace it with correct url.
+ * We now that file is in importmap because aceEditor load only filename and in immportmap we have full path to file.
+ */
+if (typeof AceWorker !== 'undefined') {
+	class AceWorker extends Worker {
+		constructor(moduleScriptUrl) {
+			const importMapScript = document.querySelector('script[type="importmap"]');
+			const importMap = JSON.parse(importMapScript.textContent);
+			let type = 'script';
+			for (const [key, url] of Object.entries(importMap.imports)) {
+				let parts = key.split('/');
+				if (parts.length > 1) {
+					let scriptName = parts[parts.length - 1];
+					if (scriptName === moduleScriptUrl) {
+						moduleScriptUrl = url;
+						type = 'module';
+						break;
+					}
+				}
+			}
+			super(moduleScriptUrl, {
+				type: type,
+			});
+		}
+	}
+
+	Worker = AceWorker;
```

Comment:
what's the purpose of doing this?

Reply:
there is comment above. that block. we need oweride global Worker